.github/workflows/build.yml

```diff
@@ -0,0 +1,30 @@
+name: build
+
+on:
+  push:
+    branches: main
+  pull_request:
+    branches: main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go-version: [1.17, 1.18]
+      fail-fast: true
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up Go ${{ matrix.go-version }} environment
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go-version }}
+          check-latest: true
+      - name: Run unit tests
+        run: make test
```

Comment:
I'm curious why the steps across [notation](https://github.com/notaryproject/notation/blob/main/.github/workflows/build.yml#L23-L34), [notation-go](https://github.com/notaryproject/notation-go/pull/60/files#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88R25-R33), and notation-core-go are slightly different, can they all be the same set of steps. Can they be [reused workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows#reusable-workflows-and-starter-workflows) defined in a single place?

Reply:
Good idea!

Reply:
Defined a reusable workflow.